Load cached images for councillors when possible

equivalentideas opened the pull request:
This adds a little logic to `Authority#load_councillors` so that we store the url for the cached S3 image of a councillor if it's available. Closes  #969

If the image isn't available, then it falls back to the whatever is stored in the popolo source data.

Could you please review this for me @henare ?

henare commented:
Nice work getting through this today @equivalentideas :+1: 

>If the image isn't available, then it falls back to the whatever is stored in the popolo source data.

I'm not sure this is the best approach. If we use URLs from the source data here and there then PlanningAlerts will be using inconsistent URLs and ones we don't control.

I'd like to consistently know that we're using images from a good, reliable source so I think we should change this to only use URLs from our cache.

henare commented on app/models/authority.rb:
```diff
@@ -261,7 +261,16 @@ def load_councillors(popolo)
+
+      councillor.popolo_id = person.id
+
+      councillor.image_url = if councillor.cached_image_available?
```
Using the ternary operator here will be more concise and clear. Of course if my suggestion about not falling back to the popolo url is accepted then this doesn't matter :smile:

henare commented on app/models/authority.rb:
```diff
@@ -261,7 +261,16 @@ def load_councillors(popolo)
+        person.image
+      end
+
+      councillor.update(email: person.email, party: person.party)
```
Will this also persist a change from setting the `popolo_id` above?

equivalentideas replied:
Yep, like the `image_url` being assigned. This is because the [#update method](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update) just calls [#save](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-save) on the record, persisting all changed attributes.

henare replied:
Shiny, I didn't know that - thanks for checking :sunglasses:

henare commented on app/models/councillor.rb:
```diff
@@ -13,4 +13,12 @@ def prefixed_name
+  end
+
+  def cached_image_available?
+    Net::HTTP.get_response(URI(cached_image_url)).kind_of? Net::HTTPSuccess
```
To avoid pulling the whole image down we should do a HEAD request instead.

equivalentideas commented:
> If we use URLs from the source data here and there then PlanningAlerts will be using inconsistent URLs and ones we don't control.

> I'd like to consistently know that we're using images from a good, reliable source so I think we should change this to only use URLs from our cache.

That sounds really good to me @henare I'm make that change.

equivalentideas commented:
I've updated this to only save an image_url if we have a cached image. I agree this is much nicer for us.

I've also changed the checking method to only do a HTTP HEAD request like you suggested @henare , a nice improvement 👍 .

Did you have any other feedback on this @henare or do you think we can merge?

henare commented:
Awesome, great work! :guitar: Can I leave this for you to deploy? :rocket:
